### DISCUSSION TITLE
Access Herd sites over local network

### DISCUSSION THREAD

**Discussion by mstowey** [beyondcode/herd-community#212] [Q&A]
Is there some way I can set up my Herd sites so I can access them over my local network without using ngrok or Expose? By IP address would be fine

**Comment by miagg** [comment]
The only way I know is to use the default site but it can only serve one site.

For this you need to add the "default" property on `/Users/youruser/Library/Application Support/Herd/config/valet/config.json`

Remember herd is using valet behind the scenes, so [this](https://laravel.com/docs/10.x/valet#serving-a-default-site) should work.

Then you can access the default site by using IP address or hostname on internal network.

Please note that in order to make it work without this hack you have to mess with dnsmasq configuration or your router settings to proxy all traffic for '.test' tld to `http://127.0.0.1:60` but I don't recommend it. I usually stick with expose to do a quick test from another device (e.g. iPhone) or you may use the Xcode Simulator while developing for responsive layout.

**Comment by mstowey** [comment]
Thanks so much. I was able to get it to work along with using the following commands:

php artisan serve --host xxx.xxx.xx.xx
npm run dev -- --host xxx.xxx.xx.xx

**Reply by vipertecpro** [reply]
Just in case if someone is still wondering over this https://stackoverflow.com/a/71933881/4650866

**Comment by guoxiangke** [comment]
I have a solution [test works 2025/08/14 update]:
Edit : /Users/xxx/Library/Application Support/Herd/config/nginx/herd.conf
```
server {
    listen 0.0.0.0:80 default_server;
```

Edit : /Users/xxx/Library/Application Support/Herd/config/nginx/nginx.conf
change:
```

    include mysite.conf;

    #include "/Applications/Herd.app/Contents/Resources/config/default-sites/*.conf";
```

mysite.conf
```
server {
    listen 0.0.0.0:80;
    #server_name my.stie.com;
    server_name _;

    root /Users/guo/Herd/xbot25/;
    index index.php index.html;
    charset utf-8;
    client_max_body_size 1024M;

    location / {
        try_files $uri $uri/ /index.php?$query_string;
    }

    location ~ \.php$ {
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass unix:/Users/guo/Library/Application\ Support/Herd/valet.sock; # Herd PHP-FPM
        fastcgi_index index.php;
        include fastcgi_params;
        fastcgi_param SCRIPT_FILENAME $realpath_root$fastcgi_script_name;
        fastcgi_param PATH_INFO $fastcgi_path_info;
    }

    location ~ /\.ht {
        deny all;
    }
}

```
